stake-pool/program/src/processor.rs

```diff
@@ -1675,23 +1731,39 @@ impl Processor {
             new_pool_tokens,
         )?;
 
+        // withdraw additional lamports to the reserve
+        let additional_lamports = all_deposit_lamports
+            .checked_sub(deposit_stake_lamports)
+            .ok_or(StakePoolError::CalculationFailure)?;
+        if additional_lamports > 0 {
+            Self::stake_withdraw(
+                stake_pool_info.key,
+                validator_stake_account_info.clone(),
+                withdraw_authority_info.clone(),
+                AUTHORITY_WITHDRAW,
+                stake_pool.withdraw_bump_seed,
+                reserve_stake_account_info.clone(),
+                clock_info.clone(),
+                stake_history_info.clone(),
+                stake_program_info.clone(),
+                additional_lamports,
+            )?;
+        }
+
```

Comment:
Aaah, so we return stake pool tokens for the full deposit, but if some part of the deposit was not staked, we put that in the reserve?

That's quite neat for handling deposits, but what happens if some joker transfers additional SOL to a validator stake account?

Reply:
You're right, we really need the mechanism during update to take out additional SOL from the validator stake accounts.